### PR TITLE
Add Dota 2 icon

### LINKED ISSUE #10567: Request: Dota 2
### Brand Name

Dota2

### Website

https://www.dota2.com

### Popularity Metric

SimilarWeb rank is 36,750 : https://www.similarweb.com/website/dota2.com/

### Official Resources for Icon and Color

PNG logo: https://cdn.cloudflare.steamstatic.com/apps/dota2/images/dota_react/global/dota2_logo_horiz.png
Press kit from Dota2 employee: https://www.reddit.com/r/DotA2/comments/249zsi/comment/ch54d19
Color: `#bf2e1a` from website's footer color logo https://cdn.cloudflare.steamstatic.com/apps/dota2/images/dota_react/dota_footer_logo.png

### Additional Comments

Moba game which inspired League of Legends (icon already available).
Has large e-sports presence and player-base: top3 most played game on steamcharts.com

### PR DESCRIPTION
**Pull request by ryanmurphyNSCC**
**Issue:** closes #10567

**Similarweb rank:** 36,750

### Checklist

- [x] I updated the JSON data in `_data/simple-icons.json`
- [x] I optimized the icon with SVGO or SVGOMG
- [x] The SVG `viewbox` is `0 0 24 24`

### Description

  The hex value is the main colour of the logo
  I did not manually vectorize the logo, I got the SVG of the logo and isolated it using BoxySVG and then optimized in SVGO.
  Only used one source, the one listed in the json data file.

### CODE REVIEW

File: _data/simple-icons.json [hunk 1]
```diff
@@ -4129,6 +4129,11 @@
             "hex": "FF3008",
             "source": "https://www.doordash.com/about/"
         },
+        {
+            "title": "Dota 2",
```
Comment (service-paradis) [inline _data/simple-icons.json:4133]: Do you know if the first version of the game used the same logo? If so, we could remove the number to keep the title as "Dota" only. We could also add an "aka" attribute with "Defense of the Ancients".

File: icons/dota2.svg [hunk 2]
```diff
@@ -0,0 +1 @@
....537 23.369 19.206 23.391 C 18.959 23.406 18.789 23.394 18.644 23.351 C 18.436 23.292 18.436 23.292 18.087 23.453 L 17.735 23.612 L 16.435 23.612 C 15.066 23.61 14.784 23.632 13.841 23.807 C 13.189 23.927 12.96 23.922 11.819 23.785 L 11.217 23.711 L 10.796 23.807 C 10.562 23.861 10.326 23.903 10.27 23.901 C 10.202 23.901 10.046 23.8 9.819 23.607 L 9.817 23.607 Z M 7.284 19.918 C 8.289 19.539 9.118 19.223 9.123 19.216 C 9.136 19.202 5.206 15.39 4.684 14.909 C 4.539 14.775 4.409 14.672 4.399 14.682 C 4.39 14.693 4.07 15.547 3.69 16.577 C 3.27 17.715 3.011 18.476 3.032 18.508 C 3.057 18.55 5.424 20.599 5.451 20.605 C 5.454 20.605 6.279 20.296 7.284 19.918 Z M 20.796 17.829 C 21.319 16.564 21.732 15.519 21.718 15.504 C 21.699 15.483 11.888 8.858 5.02 4.226 L 4.469 3.855 L 3.569 4.262 C 3.072 4.484 2.673 4.687 2.68 4.709 C 2.69 4.737 6.011 8.219 10.064 12.449 L 17.432 20.141 L 18.638 20.133 L 19.847 20.125 L 20.796 17.829 Z M 19.434 6.972 C 19.607 6.01 19.75 5.191 19.75 5.151 C 19.75 5.108 19.497 4.902 19.074 4.605 C 18.702 4.343 18.373 4.115 18.344 4.096 C 18.291 4.063 14.442 5.104 14.447 5.149 C 14.451 5.179 19.091 8.748 19.107 8.731 C 19.115 8.725 19.261 7.932 19.434 6.972 Z"/></svg>
```
Comment (service-paradis) [inline icons/dota2.svg:1]: I'm seeing some difference comparing to the original. Ex.:
image

File: _data/simple-icons.json [hunk 3]
```diff
@@ -4129,6 +4129,11 @@
...
+            "hex": "BF2E1A",
+            "source": "https://en.m.wikipedia.org/wiki/File:Dota_logo.svg"
```
Comment (service-paradis) [inline _data/simple-icons.json:4135]: We can use the Wikimedia version which is the source of the Wikipedia one:
```suggestion
            "source": "https://commons.wikimedia.org/wiki/File:Dota_logo.svg"
```

**Review by service-paradis — CHANGES_REQUESTED** [review]
Thank you @ryanmurphyNSCC for your contribution. I left a few comments if you can take a look!

File: _data/simple-icons.json (hunk 1 above)
Comment (gmse) [reply _data/simple-icons.json:4133]: This logo is only used for Valve's Dota 2. The Warcraft3 mod used a different one.

The full title "Defense of the Ancients" is usually associated with the first version, although Valve also owns the trademark. ([source](https://liquipedia.net/dota2/Defense_of_the_Ancients)).

**Review by adamrusted — APPROVED** [review]
LGTM! Thanks @ryanmurphyNSCC

**Pull request merged**